smart-contracts/contracts/Unlock.sol

```diff
@@ -200,6 +203,15 @@ contract Unlock is
     emit UnlockTemplateAdded(impl, version);
   }
 
+  /**
+  * Registers the address of the migration script to use during upgrade
+  * @param _migrationScript the address of the contract
+  * @dev if set to address zero, migration step will be ignore during upgrade
+  */
+  function setMigrationScript(address _migrationScript) public onlyOwner {
```

Comment:
Hum. We will need migration scripts for each version :/ Can we move the migrationScript address inside the PublicLock?